### DISCUSSION TITLE
RFP: Full Text Search

### DISCUSSION THREAD

**Discussion by OmarMcAdam** [un/inbox#16] [General]
we dont yet have a way to search for content in emails
luckily each organizations data has the orgId attached

ideally we would index the following table columns on a per org access basis:
`convoEntries.bodyPlainText`
`convoSubjects.subject`

when searching through the index, we need to ensure the searching user has access to all the data, likely we'll need to save an array of userIds who have access to it or we can filter the results server side before responding to the user

i am opening the door to proposals and ideas on how we can accomplish this
very interested in lessons learned from previous implementations and things we need to look out for

**Comment by me-imfhd** [comment]
Pardon me if get that incorrect, if we are trying to search contents in emails. My proposal will be we can use text-search vector databases, i guess we'll have to make a query, narrowing with orgId and contents user has access to, and return that data, very accurate data back matching the texts user is searching for.
Thank you.

PS: I would love to know what are we searching the contents in the emails for?

**Reply by OmarMcAdam** [reply]
> PS: I would love to know what are we searching the contents in the emails for?

if users want to find a specific email in their account

after some discussions on twitter we may initially implement standard mysql text search, then later add meilisearch
is that along the lines of what you were thinking?

**Reply by me-imfhd** [reply]
Yeah, this sounds like a good plan!
